manifests/init.pp

```diff
@@ -157,50 +178,80 @@
-  Hash                           $checks                       = {},
-  Hash                           $config_defaults              = $consul::params::config_defaults,
-  Stdlib::Absolutepath           $config_dir                   = $consul::params::config_dir,
-  Hash                           $config_hash                  = {},
-  String[1]                      $config_mode                  = '0664',
-  String[1]                      $docker_image                 = 'consul',
-  String[1]                      $download_extension           = 'zip',
-  Optional[Stdlib::HTTPUrl]      $download_url                 = undef,
-  String[1]                      $download_url_base            = 'https://releases.hashicorp.com/consul/',
-  Array                          $extra_groups                 = [],
-  Optional[String[1]]            $extra_options                = undef,
-  String[1]                      $group                        = $consul::params::group,
-  Stdlib::Absolutepath           $log_file                     = '/var/log/consul',
-  String[1]                      $init_style                   = $consul::params::init_style,
-  String[1]                      $install_method               = 'url',
-  Optional[String[1]]            $join_wan                     = undef,
-  Boolean                        $manage_group                 = $consul::params::manage_group,
-  Boolean                        $manage_service               = true,
-  Boolean                        $manage_user                  = $consul::params::manage_user,
-  String[1]                      $os                           = $facts['kernel'].downcase,
-  String[1]                      $package_ensure               = 'latest',
-  String[1]                      $package_name                 = 'consul',
-  Boolean                        $pretty_config                = false,
-  Integer                        $pretty_config_indent         = 4,
-  Optional[Stdlib::HTTPUrl]      $proxy_server                 = undef,
-  Boolean                        $purge_config_dir             = true,
-  Boolean                        $restart_on_change            = true,
-  Boolean                        $service_enable               = true,
-  Enum['stopped', 'running']     $service_ensure               = 'running',
-  Hash                           $services                     = {},
-  String[1]                      $user                         = $consul::params::user,
-  String[1]                      $version                      = '1.2.3',
-  Hash                           $watches                      = {},
-  Optional[String[1]]            $shell                        = $consul::params::shell,
-  Boolean                        $enable_beta_ui               = false,
-  Boolean                        $allow_binding_to_root_ports  = false,
+  Hash                           $acls                        = {},
+  Hash[String[1], Struct[{
```

Comment:
can you please put those types into a custom datatype? That would make the file more readable.

Reply:
Good idea. Replaced by type aliases.